cpp/src/gandiva/llvm_generator.cc

```diff
@@ -93,19 +127,80 @@ Status LLVMGenerator::Build(const ExpressionVector& exprs) {
   return Status::OK();
 }
 
+Status LLVMGenerator::BuildForAllMode(const ExpressionVector& exprs) {
+  Status status;
+  for (auto& expr : exprs) {
+    auto output = annotator_.AddOutputFieldDescriptor(expr->result());
+    status = AddForAllMode(expr, output);
+    GANDIVA_RETURN_NOT_OK(status);
+  }
+  // optimise, compile and finalize the module
+  status = engine_->FinalizeModule(optimise_ir_, dump_ir_);
+  GANDIVA_RETURN_NOT_OK(status);
+  // setup the jit functions for each expression.
+  for (auto& compiled_expr : compiled_exprs_) {
+    llvm::Function* ir_func = compiled_expr->ir_function();
+    EvalFunc fn = reinterpret_cast<EvalFunc>(engine_->CompiledFunction(ir_func));
+    compiled_expr->set_jit_function(fn);
+    llvm::Function* ir_func_sel_vec_16 = compiled_expr->ir_function_sel_vec_16();
+    EvalFunc_Sel_Vec_16 fn_sel_vec_16 = reinterpret_cast<EvalFunc_Sel_Vec_16>(
+        engine_->CompiledFunction(ir_func_sel_vec_16));
+    compiled_expr->set_jit_function_sel_vec_16(fn_sel_vec_16);
+    llvm::Function* ir_func_sel_vec_32 = compiled_expr->ir_function_sel_vec_32();
+    EvalFunc_Sel_Vec_32 fn_sel_vec_32 = reinterpret_cast<EvalFunc_Sel_Vec_32>(
+        engine_->CompiledFunction(ir_func_sel_vec_32));
+    compiled_expr->set_jit_function_sel_vec_32(fn_sel_vec_32);
+  }
+
+  return Status::OK();
+}
+
 /// Execute the compiled module against the provided vectors.
 Status LLVMGenerator::Execute(const arrow::RecordBatch& record_batch,
                               const ArrayDataVector& output_vector) {
+  arrow::Buffer* null_buffer = NULL;
+  int mode = -1;
+  return Execute(record_batch, mode, *null_buffer, output_vector);
+}
+
+/// Execute the compiled module against the provided vectors based on the type of
+/// selection veector.
```

Comment:
veector -> vector

Reply:
resolved it. Thank you for pointing it out.